[issue_tracker] Error given when updating issue

**Describe the bug**
When I add a comment to an issue and click "Submit" I get the following error swal: 
![image](https://user-images.githubusercontent.com/51176779/143052096-b66e39f8-5891-4310-be97-33dd5ef97b6c.png)

as well as the following in my error_log:
`[Tue Nov 23 10:13:01.880401 2021] [php7:notice] [pid 31784] [client ::1:60308] [ERROR] The server is not configured to receive a file this large., referer: http://localhost:8080/issue_tracker/issue/25`

However, when I refresh, the issue is updated with my new comment. 

I'm not sure what the error in the error log is referring to because there was no file / attachment in the issue. 

**To Reproduce**
Steps to reproduce the behavior (attach screenshots if applicable):
1. Go to issue tracker
2. Click on any issue
3. Enter a new comment
4. Click "Submit"
5. See error

**What did you expect to happen?**
The issue to be successfully updated with a success swal.

--- a/modules/issue_tracker/jsx/IssueForm.js
+++ b/modules/issue_tracker/jsx/IssueForm.js
@@ -409,11 +409,12 @@ class IssueForm extends Component {
   handleSubmit(e) {
     e.preventDefault();
 
-    // Prevent new issue submissions while one is already in progress
-    if (this.state.submissionResult && this.state.isNewIssue) return;
-    this.setState({submissionResult: 'pending'});
-
-    const myFormData = this.state.formData;
+    const state = Object.assign({}, this.state);
+    // issue submissions already in progress
+    if (state.submissionResult && state.isNewIssue) {
+      return;
+    }
+    const myFormData = state.formData;
     const formRefs = this.refs;
     const formData = new FormData();
 
@@ -421,6 +422,9 @@ class IssueForm extends Component {
     if (!this.isValidForm(formRefs, myFormData)) {
       return;
     }
+
+    // Prevent multiple submissions
+    this.setState({submissionResult: 'pending'});
 
     for (let key in myFormData) {
       if (myFormData[key] !== '') {

--- a/modules/issue_tracker/jsx/IssueForm.js
+++ b/modules/issue_tracker/jsx/IssueForm.js
@@ -402,7 +402,7 @@ class IssueForm extends Component {
   }
 
   /**
-   * Handles form submission
+   * Handles form submission for new issue being created
    *
    * @param {event} e form submit event
    */

--- a/modules/issue_tracker/jsx/IssueForm.js
+++ b/modules/issue_tracker/jsx/IssueForm.js
@@ -441,7 +441,7 @@ class IssueForm extends Component {
         response.json().then((data) => {
           this.setState({submissionResult: 'error'});
           let msgType = 'error';
-          let message = data.message || 'Failed to submit issue :(';
+          const message = data.error ?? data.message;
           this.showAlertMessage(msgType, message);
         });
         return;

--- a/modules/issue_tracker/php/edit.class.inc
+++ b/modules/issue_tracker/php/edit.class.inc
@@ -458,7 +458,7 @@ class Edit extends \NDB_Page implements ETagCalculator
 
         // Attachment for new issue.
         $uploadedFiles = $request->getUploadedFiles();
-        if (isset($uploadedFiles)) {
+        if (isset($uploadedFiles) && !empty($uploadedFiles)) {
             $attachment = new \LORIS\issue_tracker\UploadHelper();
             $success    = $attachment->setupUploading(
                 $user,

--- a/modules/issue_tracker/php/edit.class.inc
+++ b/modules/issue_tracker/php/edit.class.inc
@@ -433,6 +433,9 @@ class Edit extends \NDB_Page implements ETagCalculator
         $issueValues['lastUpdatedBy'] = $user->getUsername();
 
         $validatedInput = $this->validateInput($validateValues, $user);
+        if (!is_array($validatedInput)) { // Error exists.
+            return $validatedInput;
+        }
         if (array_key_exists('sessionID', $validatedInput)) {
             $issueValues['sessionID'] = $validatedInput['sessionID'];
         }
@@ -473,7 +476,7 @@ class Edit extends \NDB_Page implements ETagCalculator
                     \Psr\Log\LogLevel::ERROR,
                     $attachment->errorMessage
                 );
-                return new \LORIS\Http\Response\JSON\InternalServerError(
+                return new \LORIS\Http\Response\JSON\Conflict(
                     $attachment->errorMessage
                 );
             }
@@ -623,7 +626,7 @@ class Edit extends \NDB_Page implements ETagCalculator
      * @param array $values values to be validated
      * @param \User $user   the user requesting the change
      *
-     * @return array|\LORIS\Http\Response\JSON\InternalServerError
+     * @return \LORIS\Http\Response\JSON\Conflict
      */
     function validateInput(array $values, \User $user)
     {
@@ -682,10 +685,10 @@ class Edit extends \NDB_Page implements ETagCalculator
             }
             if (!$validCenter) {
                 $this->logger->log(
-                    \Psr\Log\LogLevel::ERROR,
+                    \Psr\Log\LogLevel::WARNING,
                     'PSCID and Center ID do not match a valid session!'
                 );
-                return new \LORIS\Http\Response\JSON\InternalServerError(
+                return new \LORIS\Http\Response\JSON\Conflict(
                     'PSCID and Center ID do not match a valid session!'
                 );
             }
@@ -710,7 +713,7 @@ class Edit extends \NDB_Page implements ETagCalculator
                     \Psr\Log\LogLevel::ERROR,
                     'PSCID and Visit Label do not match a valid candidate session!'
                 );
-                return new \LORIS\Http\Response\JSON\InternalServerError(
+                return new \LORIS\Http\Response\JSON\Conflict(
                     'PSCID and Visit Label do not match a valid candidate session!'
                 );
             }
@@ -735,7 +738,7 @@ class Edit extends \NDB_Page implements ETagCalculator
                     \Psr\Log\LogLevel::ERROR,
                     'PSCID does not match a valid candidate!'
                 );
-                return new \LORIS\Http\Response\JSON\InternalServerError(
+                return new \LORIS\Http\Response\JSON\Conflict(
                     'PSCID does not match a valid candidate!'
                 );
             }
@@ -749,7 +752,7 @@ class Edit extends \NDB_Page implements ETagCalculator
                 \Psr\Log\LogLevel::ERROR,
                 'Visit Label must be accompanied by a PSCID'
             );
-            return new \LORIS\Http\Response\JSON\InternalServerError(
+            return new \LORIS\Http\Response\JSON\Conflict(
                 'Visit Label must be accompanied by a PSCID'
             );
         }

--- a/modules/issue_tracker/php/edit.class.inc
+++ b/modules/issue_tracker/php/edit.class.inc
@@ -458,7 +458,7 @@ class Edit extends \NDB_Page implements ETagCalculator
 
         // Attachment for new issue.
         $uploadedFiles = $request->getUploadedFiles();
-        if (isset($uploadedFiles) && !empty($uploadedFiles)) {
+        if (!empty($uploadedFiles)) {
             $attachment = new \LORIS\issue_tracker\UploadHelper();
             $success    = $attachment->setupUploading(
                 $user,

--- a/modules/issue_tracker/php/edit.class.inc
+++ b/modules/issue_tracker/php/edit.class.inc
@@ -568,9 +568,14 @@ class Edit extends \NDB_Page implements ETagCalculator
 
         if (isset($changed_assignee)) {
             $issueChangeEmailsAssignee = $db->pselect(
-                "SELECT u.Email as Email, u.First_name as firstname " .
-                "FROM users u WHERE u.UserID=:assignee
-                AND u.UserID<>:currentUser",
+                "SELECT
+                            u.Email AS Email,
+                            u.First_name AS firstname
+                        FROM
+                            users u
+                        WHERE
+                            u.UserID = :assignee
+                            AND u.UserID != :currentUser",
                 [
                     'assignee'    => $changed_assignee,
                     'currentUser' => $user->getUserName(),
@@ -597,11 +602,11 @@ class Edit extends \NDB_Page implements ETagCalculator
                         u.First_name AS firstname
                     FROM
                         users u
-                        INNER JOIN issues_watching w ON (w.userID = u.userID)
+                        INNER JOIN issues_watching w ON (w.userID = u.UserID)
                     WHERE
                         w.issueID = :issueID
-                        AND u.UserID <> :uid
-                        AND u.UserID <> :assignee",
+                        AND w.userID = :uid
+                        AND u.UserID = :assignee",
             [
                 'issueID'  => $issueID,
                 'uid'      => $user->getUsername(),

--- a/modules/issue_tracker/php/edit.class.inc
+++ b/modules/issue_tracker/php/edit.class.inc
@@ -626,7 +626,7 @@ class Edit extends \NDB_Page implements ETagCalculator
      * @param array $values values to be validated
      * @param \User $user   the user requesting the change
      *
-     * @return \LORIS\Http\Response\JSON\Conflict
+     * @return \LORIS\Http\Response\JSON\Conflict|array
      */
     function validateInput(array $values, \User $user)
     {

--- a/modules/issue_tracker/php/edit.class.inc
+++ b/modules/issue_tracker/php/edit.class.inc
@@ -2,7 +2,6 @@
 
 namespace LORIS\issue_tracker;
 
-use Email;
 use LORIS\issue_tracker\Provisioners\AttachmentProvisioner;
 use \Psr\Http\Message\ServerRequestInterface;
 use \Psr\Http\Message\ResponseInterface;
@@ -585,7 +584,7 @@ class Edit extends \NDB_Page implements ETagCalculator
             if (isset($issueChangeEmailsAssignee[0])) {
                 $msg_data['firstname'] = $issueChangeEmailsAssignee[0]['firstname'];
 
-                Email::send(
+                \Email::send(
                     $issueChangeEmailsAssignee[0]['Email'],
                     'issue_assigned.tpl',
                     $msg_data
@@ -621,7 +620,7 @@ class Edit extends \NDB_Page implements ETagCalculator
 
         foreach ($issue_change_emails as $email) {
             $msg_data['firstname'] = $email['firstname'];
-            Email::send($email['Email'], 'issue_change.tpl', $msg_data);
+            \Email::send($email['Email'], 'issue_change.tpl', $msg_data);
         }
     }
 

